src/main/java/org/minions/devfund/lourdes/MovieDatabase.java

```diff
@@ -0,0 +1,129 @@
+package org.minions.devfund.lourdes;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Class that manage the movie information.
+ */
+public class MovieDatabase {
+    private List<Movie> movieList;
+    private List<Actor> actorList;
+
+    /**
+     * Constructor.
+     */
+    public MovieDatabase() {
+        this.movieList = new ArrayList<>();
+        this.actorList = new ArrayList<>();
+    }
+
+    /**
+     * Method to add a new movie to the movie list.
+     *
+     * @param name   represents the movie name.
+     * @param actors actor list from a movie.
+     */
+    public void addMovie(final String name, final String[] actors) {
+        Movie movie = createMovie(name);
+        ArrayList<Actor> movieActors = new ArrayList<>();
+        for (String actorName : actors) {
+            Actor actor = createActor(actorName);
+            actor.addMovie(movie);
+            movieActors.add(actor);
+        }
+        movie.setActors(movieActors);
+    }
+
+    /**
+     * Method that create a new new movie object.
+     *
+     * @param movieName represents the movie name.
+     * @return return the created movie.
+     */
+    public Movie createMovie(final String movieName) {
+        for (Movie movie : movieList) {
+            if (movie.getName().equals(movieName)) {
+                return movie;
+            }
+        }
+        Movie movie = new Movie(movieName);
+        movieList.add(movie);
+        return movie;
+    }
+
+    /**
+     * Method that create a new Actor object.
+     *
+     * @param actorName represents the actor name to be created.
+     * @return the created actor.
+     */
+    public Actor createActor(final String actorName) {
+        for (Actor actor : actorList) {
+            if (actor.getName().equals(actorName)) {
+                return actor;
+            }
+        }
+        Actor newActor = new Actor(actorName);
+        actorList.add(newActor);
+        return newActor;
+    }
+
+    /**
+     * Method that update the rating of the given movie name.
+     *
+     * @param name      represents the movie name.
+     * @param newRating the new rating to be set.
+     */
+    public void updateRating(final String name, double newRating) {
+        movieList.stream()
+                .filter(movie -> movie.getName().equals(name))
+                .findFirst()
+                .get()
+                .setRating(newRating);
+    }
+
+    /**
+     * Method that return the best Actor.
+     *
+     * @return the best actor.
+     */
+    public String getBestActor() {
+        return actorList.stream()
+                .max(Comparator.comparingDouble(Actor::averageRating))
+                .get()
+                .getName();
+    }
+
+    /**
+     * Method that return the best movie fo the list.
+     *
+     * @return the best movie.
+     */
+    public String getBestMovie() {
+        return movieList.stream()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)